### ISSUE TITLE
[Feature Request] Rename stashes

### ISSUE THREAD

**Issue by ocornut** [ForkIssues/TrackerWin#602]
I realize I'm coming off as obsessed with renaming :)

On my wishlist - and again, I'm not sure how fits with git workflow and how realistic it is to implement - but I would love if it was possible to rename a __stash__ while preserving its order in the stash list.

Thank you for reading!

**Comment by jerone** [comment]
I too want to rename my stashes. The name I gave it doesn't really describe what it contains. Also, sometimes Fork creates an "autostash", which is useful but the name is not.

**Comment by DanPristupov** [comment]
I don't know a reliable way to rename a stash.

**Comment by jerone** [comment]
I [research a bit](https://stackoverflow.com/questions/25931026/how-can-i-rename-a-git-stash) after your answer, and there is an way: https://stackoverflow.com/a/42671046/108448

But as stated in that answer, there are some drawbacks:
1. This switches branches temporarily.
2. It renumbers the stashes, so the changed stash becomes `stash@{0}`.

Point 2 should not be an issue, as people are already using an GUI.
Point 1 is interesting. Aldo it should not matter, because the branch is created and then after deleted again.

(The other drawbacks mentioned in the [answer](https://stackoverflow.com/a/42671046/108448) are no problem for an GUI.)

**Comment by DanPristupov** [comment]
We just released Fork 1.58 which allows renaming stashes.

There are some drawbacks caused by the immutable nature of git (e.g. renamed stash becomes first in the stash list). But after all, it works pretty well and doesn't require a clean working directory or branch switch.

It's still a new feature, so use it with a bit of caution, as the dialog advises :)

<img width="550" alt="Screenshot 2021-01-15 at 17 31 59" src="https://user-images.githubusercontent.com/618115/104752923-93e59500-5757-11eb-9641-ee38ae9fc551.png">

<img width="400" alt="Screenshot 2021-01-15 at 17 42 46" src="https://user-images.githubusercontent.com/618115/104754065-1753b600-5759-11eb-98cf-3e6ee969a947.png">

**Comment by ocornut** [comment]
This seems to be working well for me with 1.58. Thank you very much @DanPristupov !

**Comment by kimsey0** [comment]
@DanPristupov: Thanks for this feature. Any chance it can have F2 added as a keyboard shortcut, like for branches?

**Comment by DanPristupov** [comment]
Yep. Tomorrow.

**Comment by pmhpereira** [comment]
@DanPristupov, I'm on 1.62 and still not working with F2. 😄

**Comment by DanPristupov** [comment]
When I press F2 on a stash in the commit list, it works for me.

**Comment by pmhpereira** [comment]
Oh, nice!
I was trying to do it on the Left Panel, but that doesn't seem to work.
![image](https://user-images.githubusercontent.com/886261/121874929-bfb57880-ccff-11eb-9791-5aaa8a4997c0.png)

Also, the right-click menu doesn't show F2 as a possible shortcut:
![image](https://user-images.githubusercontent.com/886261/121875054-ebd0f980-ccff-11eb-8941-cc44a20d0f26.png)

**Comment by xmedeko** [comment]
Yeah, F2 on the left pane to rename the stash would be nice, too.

**Comment by ocornut** [comment]
I just noticed that if I include a `:` character in the new stash name it can behaves oddly, leaving only what's after the `:`, e.g. "hello:world" would turn the stash into "world". I also think in one instance it prevented the rename alltogether but I can't repro it.

It's probably a git restriction in which case ideally the dialog box would filter them or prevent from making that.

Oddly enough your original screenshot https://github.com/fork-dev/TrackerWin/issues/602#issuecomment-761051317 had `:` in the message :)

Not so important but best written somewhere!
